Backend/src/main/java/org/tah/service/Service.java

```diff
@@ -0,0 +1,87 @@
+package org.tah.service;
+
+import org.tah.model.Task;
+import org.tah.model.TaskStatusEnum;
+import org.tah.persistence.TaskDAO;
+
+import java.sql.SQLException;
+
+public class Service {
+
+    private TaskDAO taskDAO = new TaskDAO();
+
+    public void addTask(Task task) {
+        validateTaskWhenAdding(task);
+        try {
+            taskDAO.insert(task);
+        } catch (SQLException e) {
+            throw new ServiceException("Failed to insert a task", e);
+        }
+    }
+
+    public void updateTask(Task task) {
+        validateTaskWhenUpdating(task);
+        try {
+            taskDAO.update(task);
+        } catch (SQLException e) {
+            throw new ServiceException("Failed to update a task.", e);
+        }
+    }
+
+    public void deleteTask(Task task) {
+        try {
+            taskDAO.delete(task);
+        } catch (SQLException e) {
+            throw new ServiceException("Failed to delete a task.", e);
+        }
+    }
+
+    public Task query(int id) {
```

Comment:
```suggestion
    public Task getTask(int id) {
```
Follow the same naming pattern as for other methods